src/Functions/Lemmatizers.cpp

```diff
@@ -0,0 +1,93 @@
+#include <Common/Exception.h>
+#include <Functions/Lemmatizers.h>
+#include <RdrLemmatizer.h>
+
+#include <vector>
+#include <filesystem>
+
+namespace DB
+{
+
+namespace ErrorCodes
+{
+    extern const int LOGICAL_ERROR;
+    extern const int UNKNOWN_ELEMENT_IN_CONFIG;
+    extern const int INVALID_CONFIG_PARAMETER;
+}
+
+
+class Lemmatizer
+{
+private:
+    RdrLemmatizer lemmatizer;
+
+public:
+    using TokenPtr = std::shared_ptr<char []>;
+
+    Lemmatizer(const String & path) : lemmatizer(path.data()) {}
+
+    TokenPtr lemmatize(const char * token)
+    {
+        return TokenPtr(lemmatizer.Lemmatize(token));
+    }
+};
+
+/// Duplicate of code from StringUtils.h. Copied here for less dependencies.
+static bool startsWith(const std::string & s, const char * prefix)
+{
+    return s.size() >= strlen(prefix) && 0 == memcmp(s.data(), prefix, strlen(prefix));
+}
+
+Lemmatizers::Lemmatizers(const Poco::Util::AbstractConfiguration & config)
+{
+    String prefix = "lemmatizers";
+    Poco::Util::AbstractConfiguration::Keys keys;
+
+    if (!config.has(prefix))
+        throw Exception("You should specify list of lemmatizers in " + prefix,
```

Comment:
Seems like the user how used this `lemmatize`  function in his query will receive this error? Maybe it would be better to say something like `No lemmatizers available` or something like this?

Reply:
This exception seems more informative to me, you know straight away what to do to get this fixed. But if needed I can change that to suggested text.

Reply:
In most cases, user who runs SQL queries don't have an access to the server :)